tls/s2n_client_hello.c

```diff
@@ -342,6 +342,16 @@ int s2n_client_hello_send(struct s2n_connection *conn)
     GUARD(s2n_stuffer_write_bytes(out, client_protocol_version, S2N_TLS_PROTOCOL_VERSION_LEN));
     GUARD(s2n_stuffer_copy(&client_random, out, S2N_TLS_RANDOM_DATA_LEN));
 
+    /* Generate client session id when empty so that when server sends
+     * an empty session id it is because it doesn't support session resumption
+     */
+    if (conn->session_id_len == 0 && conn->config->use_tickets) {
```

Comment:
I think RFC suggests generating a random session id each time you use session ticket, so should we generate it regardless if you have session_id_len or not?

Reply:
Ahh will do. 

Wasn't clear for me from reading the RFC, if 'may generate' meant may generate no matter what or if only in certain cases. 

> When presenting a ticket, the client MAY generate and include a Session ID in the TLS ClientHello.